### PR TITLE
Domain whitelisting for web search results

### PR DESCRIPTION
**Pull request by que-nguyen**
Note: Meta search engines like SearxNG do not support filtering results by domain natively, so this filtering is applied post-search.
Added a filter to restrict search results to specified domains.
Updated the processing of search results to apply the whitelist filter before returning the final results.
Ensured that only results from allowed domains (if specified in the whitelist) are included in the output.

### CODE REVIEW

**Comment by programminginhackerway** [comment]
> SearxNG

Thanks, does it work well? I found that SearxNG only supports the high priority option and it can't handle the unwanted websites in results.

**Comment by programminginhackerway** [comment]
> Note: Meta search engines like SearxNG do not support filtering results by domain natively, so this filtering is applied post-search. Added a filter to restrict search results to specified domains. Updated the processing of search results to apply the whitelist filter before returning the final results. Ensured that only results from allowed domains (if specified in the whitelist) are included in the output.



> > Meta search engines like SearxNG do not support filtering results by domain natively, so this filtering is applied post-search.
> 
> Thanks, does it work well? I found that SearxNG only supports the high priority option and it can't handle the unwanted websites in results.

File: backend/apps/rag/main.py [hunk 1]
```diff
@@ -112,6 +112,7 @@
     YOUTUBE_LOADER_LANGUAGE,
     ENABLE_RAG_WEB_SEARCH,
     RAG_WEB_SEARCH_ENGINE,
+    RAG_WEB_SEARCH_WHITE_LIST_DOMAINS,
```
Comment (tjbck) [inline backend/apps/rag/main.py:115]: `RAG_WEB_SEARCH_DOMAIN_FILTER_LIST`

File: backend/config.py [hunk 2]
```diff
@@ -895,6 +895,15 @@ class BannerModel(BaseModel):
     os.getenv("RAG_WEB_SEARCH_ENGINE", ""),
 )
 
+RAG_WEB_SEARCH_WHITE_LIST_DOMAINS = PersistentConfig(
+    "RAG_WEB_SEARCH_WHITE_LIST_DOMAINS",
+    "rag.rag_web_search_white_list_domains",
```
Comment (tjbck) [inline backend/config.py:900]: `rag.web.search.domain.filter_list`

File: backend/config.py [hunk 3]
```diff
@@ -895,6 +895,15 @@ class BannerModel(BaseModel):
...
+    [
+        # "example.com",
```
Comment (tjbck) [inline backend/config.py:902]: The default option should be more explicit.

File: backend/apps/rag/search/main.py [hunk 4]
```diff
@@ -1,8 +1,18 @@
 from typing import Optional
-
+from urllib.parse import urlparse
 from pydantic import BaseModel
 
 
+def filter_by_whitelist(results, whitelist):
```
Comment (tjbck) [inline backend/apps/rag/search/main.py:6]: `get_filtered_results` with `filter_list` param

File: backend/apps/rag/search/brave.py [hunk 5]
```diff
@@ -1,15 +1,15 @@
...
-from apps.rag.search.main import SearchResult
+from apps.rag.search.main import SearchResult, filter_by_whitelist
 from config import SRC_LOG_LEVELS
 
 log = logging.getLogger(__name__)
 log.setLevel(SRC_LOG_LEVELS["RAG"])
 
 
-def search_brave(api_key: str, query: str, count: int) -> list[SearchResult]:
+def search_brave(api_key: str, query: str, whitelist:List[str], count: int) -> list[SearchResult]:
```
Comment (tjbck) [inline backend/apps/rag/search/brave.py:12]: `filter_list` should be optional here.

**Comment by tjbck** [comment]
The rest LGTM!

**Comment by tjbck** [comment]
LGTM, Thanks!

**Pull request merged**